GVRf/Framework/src/org/gearvrf/GVRMesh.java

```diff
@@ -17,20 +17,26 @@
 
 import static org.gearvrf.utility.Assert.*;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import org.gearvrf.utility.Exceptions;
+import org.gearvrf.utility.Log;
 
 /**
  * This is one of the key GVRF classes: It holds GL meshes.
  * 
  * A GL mesh is a net of triangles that define an object's surface geometry.
  */
-public class GVRMesh extends GVRHybridObject {
+public class GVRMesh extends GVRHybridObject implements PrettyPrint {
     public GVRMesh(GVRContext gvrContext) {
-        super(gvrContext, NativeMesh.ctor());
+        this(gvrContext, NativeMesh.ctor());
     }
 
     GVRMesh(GVRContext gvrContext, long ptr) {
         super(gvrContext, ptr);
+        setBones(new ArrayList<GVRBone>());
```

Comment:
It seems to me like we need a special kind of a mesh - a mesh with bones, and keep this one generic.